src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -340,8 +340,8 @@ public void removeCitizen(final ICitizenData citizen)
                 optCitizen.get().setItemStackToSlot(EquipmentSlotType.OFFHAND, ItemStackUtils.EMPTY);
             }
         }
-        colony.getCitizenManager().calculateMaxCitizens();
         super.removeCitizen(citizen);
+        colony.getCitizenManager().calculateMaxCitizens();
```

Comment:
haha, that's bad

Reply:
ye before it would recalc it before actually removing :D